test/functional/test-xhr.js

```diff
@@ -268,15 +268,13 @@ describe('XHR', function() {
               });
         });
 
-        it('should not return a resolved indented promise', () => {
+        it('should not resolve after rejecting promise', () => {
           mockXhr.status = 500;
           mockXhr.responseText = '{"a": "hello"}';
           mockXhr.headers['Content-Type'] = 'application/json';
           mockXhr.getResponseHeader = () => 'application/json';
-          const promise = assertSuccess(
-              createResponseInstance('{"a": 2}', mockXhr));
-          promise.should.be.rejected;
```

Comment:
I've tested with `eventually.be.rejected` that wasn't the issue. The initial issue we thought might happen (on a previous PR #3669) is that returning the already resolved promise returned from `response.json()` inside the `new Promise` constructor would actually resolve the promise instead of actually the intended behavior of rejecting it. This doesn't see true that's why this test no longer needed/valid.